lib/focus-out-shell/component/FocusOutShell.tsx

```diff
@@ -0,0 +1,93 @@
+// Note: keep the dependencies for this generic component at a minimal (e.g. don't import OfficeFabric)
+import * as React from 'react';
+
+export interface IFocusOutShellProps {
+    className?: string,
+    onFocus?: (ev: React.FocusEvent<HTMLElement>) => void;
+    onBlur?: (ev: React.FocusEvent<HTMLElement>) => void;
+    allowMouseDown?: (target: HTMLElement) => boolean;
+}
+
+export interface IFocusOutShellChildContext {
+    callOutClassName: string,
+    callOutOnDismiss: (ev: React.FocusEvent<HTMLElement>) => void;
+}
+
+export const FocusOutShellChildContextTypes: React.ValidationMap<any> = {
+    callOutClassName: React.PropTypes.string.isRequired,
+    callOutOnDismiss: React.PropTypes.func.isRequired
+}
+
+export default class FocusOutShell extends React.PureComponent<IFocusOutShellProps, {}> {
+    public static readonly childContextTypes = FocusOutShellChildContextTypes;
+
+    private static readonly BaseClassName = "focus-out-shell";
+    private static readonly CallOutClassName = `${FocusOutShell.BaseClassName}-callout`;
+
+    private _containerDiv: HTMLDivElement;
+
+    public render(): JSX.Element {
+        const { children, onFocus, className } = this.props;
+
+        let rootClassName = FocusOutShell.BaseClassName;
+        if (className) {
+            rootClassName = `${rootClassName} ${className}`;
+        }
+
+        return (
+            <div className={rootClassName} ref={this._containerDivOnRef} onBlur={this._onBlur} onFocus={onFocus} onMouseDown={this._onMouseDown}>
+                {children}
+            </div>
+        );
+    }
+
+    public getChildContext(): IFocusOutShellChildContext {
+        return {
+            callOutClassName: FocusOutShell.CallOutClassName,
+            callOutOnDismiss: this._callOutOnDismiss
+        };
+    }
+
+    private _callOutOnDismiss = (ev: React.FocusEvent<HTMLElement>): void => {
+        // target is the event object from the document.body focus event (captured by the Callout component)
+        const nextTarget = ev.target as HTMLElement;
+
+        if (this._containerDiv && nextTarget && this._containerDiv.contains(nextTarget)) {
+            return;
+        }
+
+        requestAnimationFrame(() => this.props.onBlur(ev));
```

Comment:
Check null for onBlur since it is optional

Reply:
good catch